File: app/Models/StoreAddress.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class StoreAddress extends Model
{
    use HasFactory;

    public $timestamps = false;

    protected $fillable = [
        'store_id',
        'address',
        'city',
        'province',
        'latitude',
        'longitude'
    ];

    protected $casts = [
        'latitude' => 'decimal:7',
        'longitude' => 'decimal:7'
    ];

    public function store(): BelongsTo
    {
        return $this->belongsTo(Store::class);
    }
}

File: mobile/lib/models/store_address.dart
class StoreAddress {
  final String address;
  final String city;
  final String province;
  final double? latitude;
  final double? longitude;

  StoreAddress({
    required this.address,
    required this.city,
    required this.province,
    this.latitude,
    this.longitude,
  });

  factory StoreAddress.fromJson(Map<String, dynamic> json) {
    return StoreAddress(
      address: (json['address'] ?? '').toString(),
      city: (json['city'] ?? '').toString(),
      province: (json['province'] ?? '').toString(),
      latitude: json['latitude'] != null ? double.tryParse(json['latitude'].toString()) : null,
      longitude: json['longitude'] != null ? double.tryParse(json['longitude'].toString()) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'address': address,
      'city': city,
      'province': province,
      'latitude': latitude,
      'longitude': longitude,
    };
  }
}
